### PR TITLE
Automatic canary releases workflow based on Changesets

### PR DESCRIPTION
**Pull request by dotansimha**
Closes https://github.com/graphql/graphiql/issues/1972

### CODE REVIEW

**Review by saihaj — APPROVED** [review]

**Comment by dotansimha** [comment]
@acao ready for a review ;)

File: scripts/canary-release.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+const applyReleasePlan = require('@changesets/apply-release-plan').default;
+const { getPackages } = require('@manypkg/get-packages');
+
+function getNewVersion(version, type) {
+  const gitHash = cp
+    .spawnSync('git', ['rev-parse', '--short', 'HEAD'])
+    .stdout.toString()
+    .trim();
+
+  return semver.inc(version, `pre${type}`, true, 'alpha-' + gitHash);
```
Comment (acao) [inline scripts/canary-release.js:19]: this should say canary if it's going to be on each PR merge. don't want to create confusion

Comment (dotansimha) [reply scripts/canary-release.js:19]: Do you mean to change to `canary` tag instead of `alpha` tag?

Comment (acao) [reply scripts/canary-release.js:19]: yeah, what do you think of that?

**Comment by dotansimha** [comment]
@acao empty changesets seems to post a weird empty comment, I'll fix it now :)

File: scripts/canary-release.js (hunk 1 above)
Comment (dotansimha) [reply scripts/canary-release.js:19]: Sure! I'll change that :)

**Comment by dotansimha** [comment]
@acao ok there was a minor issue with the PR comment, but alpha releasing works perfectly now :)  It looks like that: 
image

And it uses the same comment and just updates it when the PR changes. I removed the experiment changeset file from this PR so it's ready to be merged.

**Comment by acao** [comment]
looks great! let's merge it tonight

**Comment by acao** [comment]
So wait, does this push a canary on each commit? Isn’t that a bit excessive? Don’t we want to release the canary on merge instead?

**Pull request merged**